reference_entities/enrich_a_record_with_new_attribute_type.rst

```diff
@@ -0,0 +1,443 @@
+                $normalizedValue['channel'],
+                $normalizedValue['locale'],
+                $normalizedValue['data']
+            );
+
+            return $command;
+        }
+    }
+
+Don't forget to register your factory to be recognized by our registry:
+
+.. code-block:: yaml
+
+    acme.application.factory.record.edit_simple_metric_value_command_factory:
+        class: Acme\CustomBundle\Application\Record\EditRecord\CommandFactory\EditSimpleMetricValueCommandFactory
+        tags:
+            - { name: akeneo_referenceentity.edit_record_value_command_factory }
+
+Now that we have our command, we need a specific value updater that will be able to understand this command to update a simple metric value:
+
+.. code-block:: php
+
+    <?php
+
+    namespace Acme\CustomBundle\Application\Record\EditRecord\ValueUpdater;
+
+    use Acme\CustomBundle\Application\Record\EditRecord\CommandFactory\EditSimpleMetricValueCommand;
+    use Acme\CustomBundle\Domain\Model\Record\Value\SimpleMetricData;
+    use Akeneo\ReferenceEntity\Application\Record\EditRecord\CommandFactory\AbstractEditValueCommand;
+    use Akeneo\ReferenceEntity\Application\Record\EditRecord\ValueUpdater\ValueUpdaterInterface;
+    use Akeneo\ReferenceEntity\Domain\Model\ChannelIdentifier;
+    use Akeneo\ReferenceEntity\Domain\Model\LocaleIdentifier;
+    use Akeneo\ReferenceEntity\Domain\Model\Record\Record;
+    use Akeneo\ReferenceEntity\Domain\Model\Record\Value\ChannelReference;
+    use Akeneo\ReferenceEntity\Domain\Model\Record\Value\LocaleReference;
+    use Akeneo\ReferenceEntity\Domain\Model\Record\Value\Value;
+
+    class SimpleMetricUpdater implements ValueUpdaterInterface
+    {
+        public function supports(AbstractEditValueCommand $command): bool
+        {
+            return $command instanceof EditSimpleMetricValueCommand;
+        }
+
+        public function __invoke(Record $record, AbstractEditValueCommand $command): void
+        {
+            if (!$this->supports($command)) {
+                throw new \RuntimeException('Impossible to update the value of the record with the given command.');
+            }
+
+            $attribute = $command->attribute->getIdentifier();
+            $channelReference = (null !== $command->channel) ?
+                ChannelReference::fromChannelIdentifier(ChannelIdentifier::fromCode($command->channel)) :
+                ChannelReference::noReference();
+            $localeReference = (null !== $command->locale) ?
+                LocaleReference::fromLocaleIdentifier(LocaleIdentifier::fromCode($command->locale)) :
+                LocaleReference::noReference();
+
+            $metricValue = SimpleMetricData::createFromNormalize($command->metricValue);
+
+            $value = Value::create($attribute, $channelReference, $localeReference, $metricValue);
+            $record->setValue($value);
+        }
+    }
+
+Of course, we need to register this updater to be recognized by our registry:
```

Comment:
```suggestion
Of course, we need to register this updater to be known by our registry:
```

Reply:
small remark also, it's prefered to avoid "of course", "simply", etc words in documentation because it can be really frustrating if the example doesn't work